app/controllers/sessions_controller.rb

```diff
@@ -23,4 +23,14 @@ def destroy
 
     redirect_to sign_in_user.logout_url(request), allow_other_host: true
   end
+
+  def failure
+    dfe_sign_in_uid = session.fetch("dfe_sign_in_user", {})["dfe_sign_in_uid"]
+    message = params[:message]
+    Sentry.capture_message("DSI failure with #{message} for dfe_sign_in_uid: #{dfe_sign_in_uid}")
```

Comment:
Since the `message` variable isn't really reused anywhere, I'd use the `params[:message]` directly in the Sentry message.

```suggestion
    Sentry.capture_message("DSI failure with #{params[:message]} for dfe_sign_in_uid: #{dfe_sign_in_uid}")
```